.circleci/config.yml

```diff
@@ -0,0 +1,154 @@
+---
+version: 2.1
+executors:
+  med: # 2cpu, 4G ram
+    docker:
+      - image: circleci/node:10
+    resource_class: medium
+    working_directory: ~/project
+
+commands:
+  prepare:
+    description: "Prepare"
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore cached dependencies
+          keys:
+            - deps-{{ checksum "package.json" }}-{{ .Branch }}-{{ .Revision }}
+            - deps-{{ checksum "package.json" }}
+            - deps-
```

Comment:
are you sure about this last key?

Reply:
Yes, this allows for most of the cache to be restored in the case where a single dependency has been updated.
In most cases we will be able to use most of the cache even in the case where the `package.json` file has been changed.

Reply:
See https://circleci.com/docs/2.0/caching/#restoring-cache for more details